addons/docs/src/blocks/Props.tsx

```diff
@@ -205,14 +196,17 @@ export const Props: FC<PropsProps> = (props) => {
   } = context;
 
   const { include, exclude, components } = props as ComponentsProps;
-  const { story } = props as StoryProps;
+  const { story, showComponent } = props as StoryProps;
 
   let allComponents = components;
   const main = getComponent(props, context);
 
-  if (!allComponents && main) {
-    const mainLabel = getComponentName(main);
-    allComponents = { [mainLabel]: main, ...subcomponents };
+  const mainLabel = getComponentName(main);
+  if (!allComponents) {
+    if (!story || showComponent) {
+      allComponents = { [mainLabel]: main };
+    }
+    allComponents = { ...allComponents, ...subcomponents };
```

Comment:
Didn't we just check that `allComponents` is falsy

Reply:
updated w/ slightly cleaner code. logic is same AFAIK because spreading falsey is allowed